### ISSUE TITLE
receiver/prometheus, exporter/prometheus: recognize SIGHUP to reload configurations as Prometheus server does

### ISSUE THREAD

**Issue by odeke-em** [open-telemetry/opentelemetry-collector#3794]
Prometheus' server recognizes uses a SIGHUP signal sent to it, to reload configurations when they are changed.
Essentially Prometheus says per https://prometheus.io/docs/prometheus/latest/configuration/configuration/
***Prometheus can reload configurations by sending SIGHUP or by visiting the /-/reload endpoint with —web.enable-lifecycle flag is enabled***
<img width="619" alt="Screen Shot 2021-08-06 at 11 47 15 PM" src="https://user-images.githubusercontent.com/4898263/128591326-38f64600-8cd0-46b3-b59a-dc98139b5959.png">

This means that we don't have parity with Prometheus server.

We filed this issue https://github.com/open-telemetry/wg-prometheus/issues/3 to implement parity with Prometheus server.

There is an old open issue from August 2019 (2 years ago) to support reload of configurations per https://github.com/open-telemetry/opentelemetry-collector/issues/273 for the entire collector. Perhaps we should also add the general ability for the OpenTelemetry collector to reload every configuration if given a SIGHUP signal.

**Please note that this issue isn't a release blocker**

**Comment by bogdandrutu** [comment]
I don't think this is a bug. we are not copying prometheus in this project. We do have other ways to reload config see the ConfigSource and other mechanisms.
